### ISSUE TITLE
paleomix bam_pipeline: error: unrecognized arguments: --gatk_max_threads=1 --progress_ui=running --jre_options=

### ISSUE THREAD

**Issue by Rhinogradentia** [MikkelSchubert/paleomix#37]
Hi,

thank you for porting to python3!

I immediately tried to create a singularity container.
Although I had to work around the conda environment it seems to work despite two things. The installation with pip only worked when providing `--ignore-installed ruamel-yaml`, because it was already somehow installed before (by conda create I assume). 

But the main reason I write this ticket is something else. When running the example workflow for testing I get the following error message:

```
$ singularity run paleomix_v1.3.sif bam example . 
13:41:13 INFO Copying example project to '.'
13:41:13 INFO Sucessfully saved example in './bam_pipeline'
$ cd bam_pipeline 
$ singularity run ../paleomix_v1.3.sif bam run makefile.yaml
usage: paleomix bam_pipeline [-h] [--version] command ...
paleomix bam_pipeline: error: unrecognized arguments: --gatk_max_threads=1 --progress_ui=running --jre_options== --ui_colors=on
```

I didn't change the `~/.paleomix/bam_pipeline.ini`. Do I have to add the value for the --jre_options to get it working? And if yes, what exactly goes there? The path to java?

```
[Defaults]
max_threads = 2
log_level = warning
gatk_max_threads = 1
jar_root = /root/install/jar_root
bwa_max_threads = 1
progress_ui = running
temp_root = /tmp/root/bam_pipeline
jre_options = 
bowtie2_max_threads = 1
adapterremoval_max_threads = 1
ui_colors = on


```

EDIT:
What I've noticed was the dual "=" after the jre_options, don't know where this is coming from.

The version is of course 1.3:
```
$ singularity run paleomix_v1.3.sif bam --version
paleomix bam_pipeline v1.3.0
```

**Comment by MikkelSchubert** [comment]
Hi,

Thank you for the report!

There are a couple of issues here, one of which I pushed a small bugfix for.
However, I realized that there were some other things I have to change as well, so I'll probably do another small bug-fix release tomorrow when I have time.

In the mean time, the simplest way forward for you is simply do to remove the following lines:

	* gatk_max_threads = 1
	* progress_ui = running
	* jre_options = 

This will not change the behavior of the pipeline.

However, I would also suggest changing "log_level = warning" to "log_level = info", since console output is currently tied to that option, meaning that you won't see a whole lot when you run the pipeline with "log_level = warning". I am going to change that in the next release, at which point I would suggest changing it back to "warning".

Or you could just wait for the next release, in which case you shouldn't need to do anything.

Best,
 Mikkel

**Comment by Rhinogradentia** [comment]
Thank you Mikkel - I will do as you suggested.
Best,
Nadine

**Comment by Rhinogradentia** [comment]
Hi Mikkel,
I followed your suggestion and got another error message with a further parameter - only FYI.

```
singularity run ../paleomix_v1.3.sif bam run makefile.yaml
usage: paleomix bam_pipeline [-h] [--version] command ...
paleomix bam_pipeline: error: unrecognized arguments: --ui_colors=on
```
When also deleting this one from the in file it first seemed to work but then this came:
```
singularity run ../paleomix_v1.3.sif bam run makefile.yaml
09:25:15 INFO Reading makefile 'makefile.yaml'
09:25:15 INFO Validating FASTA files
09:25:15 INFO Indexing FASTA at 'prefixes/rCRS.fasta'
09:25:15 INFO Building BAM pipeline for 'makefile.yaml'
09:25:17 INFO Running BAM pipeline
09:25:17 INFO Checking file dependencies
09:25:17 INFO Checking for auxiliary files
09:25:17 INFO Checking required software
09:25:17 INFO  - Found Rscript v3.5.1
09:25:18 INFO  - Found AdapterRemoval v2.3.1
09:25:18 INFO  - Found BWA v0.7.17
09:25:18 INFO  - Found Picard tools v2.22
09:25:19 INFO  - Found mapDamage v2.0.8
09:25:19 ERROR Version requirements not met for mapDamage: Expected at least v2.2.1, but found v2.0.8
09:25:19 INFO Logging warnings and errors to '/vagrant/Paleomix/bam_pipeline/bam_pipeline.20200903_092515_01.log'
09:25:19 INFO  - Found samtools v1.9.0
09:25:19 ERROR Please refer to the PALEOMIX installation instructions at https://paleomix.readthedocs.io/
```
It seems that the conda environment does not install the correct mapDamage version?

I installed it this way:
`./conda create -c bioconda -n paleomix python=2.7 pip adapterremoval=2.3.1 samtools=1.9 picard=2.22.9 bowtie2=2.3.5.1 bwa=0.7.17 mapdamage2=2.0.9 r-base=3.5.1 r-rcpp=1.0.4.6 r-rcppgsl=0.3.7 r-gam=1.16.1 r-inline=0.3.15`, following your docs. After exchanging the entry by `mapdamage2=2.2.1`  a lot of conflicts occur. 

```
 ./conda create -c bioconda -n paleomix python=2.7 pip adapterremoval=2.3.1 samtools=1.9 picard=2.22.9 bowtie2=2.3.5.1 bwa=0.7.17 mapdamage2=2.2.1 r-base=3.5.1 r-rcpp=1.0.4.6 r-rcppgsl=0.3.7 r-gam=1.16.1 r-inline=0.3.15
[...]
Package setuptools conflicts for:
python=2.7 -> pip -> setuptools
pip -> setuptools

Package pip conflicts for:
pip
mapdamage2=2.2.1 -> python[version='>=3.5'] -> pip
python=2.7 -> pip
bowtie2=2.3.5.1 -> python[version='>=2.7,<2.8.0a0'] -> pip
```
My Miniconda-Version is: Miniconda3-latest-Linux-x86_64.sh 

I think I will wait for next bug-fixes...

Best,
Nadine

**Comment by MikkelSchubert** [comment]
Hi Nadine,

I've released a new version of PALEOMIX (v1.3.2) that should hopefully address both of the problems that you can into.
The conda installation instructions have changed a bit, and now always use the environment file to perform the install, but they also install everything in one step.

Thank you again for the bug reports and please let me know if you run into any other problems.

Best,
 Mikkel

**Comment by Rhinogradentia** [comment]
Dear Mikkel,

this time everything ran through smoothly with the example data.

Thanks a lot!

Best, 
Nadine
